app/src/main/kotlin/edu/stanford/bdh/engagehf/MainActivity.kt

```diff
@@ -80,6 +96,25 @@ class MainActivity : FragmentActivity() {
                 }
             }
         }
+        logger.i { "Intent Data from Main Activity ${intent.extras}" }
```

Comment:
This will work in case the app did not launch yet. In case the main activity has started and the app is in background we will create a new instance of it. I propose the following. Update the `android:launchMode="singleInstance"` for the `MainActivity` in `AndroidManifest.xml`, and override here `onNewIntent` method, from the docs of `onNewIntent`

> Only allow one instance of this activity to ever be running. This activity gets a unique task with only itself running in it; if it is ever launched again with the same Intent, then that task will be brought forward and its {@link android. app. Activity#onNewIntent Activity. onNewIntent()} method called. If this activity tries to start a new activity, that new activity will be launched in a separate task. See the Tasks and Back Stack document for more details about tasks.